src/core/overlay.ts

```diff
@@ -0,0 +1,96 @@
+import * as Config from '@oclif/config';
+import debug from 'debug';
+import jsonpath from 'jsonpath';
+import mergician from 'mergician';
+import { JSONSchema4Object } from 'json-schema';
+import { safeStringify } from '@stoplight/yaml';
+
+import { APIDefinition, OpenAPIOverlay } from '../definition';
+
+export class Overlay {
+  _config: Config.IConfig;
+
+  public constructor(config: Config.IConfig) {
+    this._config = config;
+  }
+
+  public async run(
+    definition: APIDefinition,
+    overlay: OpenAPIOverlay,
+    format: string,
+  ): Promise<string | undefined> {
+    const newDefinition = this.applyOverlay(definition, overlay);
+
+    if (format == 'json') {
+      return JSON.stringify(newDefinition);
+    } else {
+      return safeStringify(newDefinition);
+    }
```

Comment:
You are right. It's because it's the first time we need to serialize the in memory tree so I didn't really think in terms of reusability yet. I'll try to suggest a change for that :+1: